pybop/models/lithium_ion/base_echem.py

```diff
@@ -290,39 +293,27 @@ def approximate_capacity(self, parameter_set: Optional[ParameterSet] = None):
         Returns
         -------
         float
-            The estimate of the nominal cell capacity.
+            The estimate of the nominal cell capacity [A.h].
         """
         parameter_set = parameter_set or self._parameter_set
 
-        # Calculate theoretical energy density
-        theoretical_energy = self._electrode_soh.calculate_theoretical_energy(
-            parameter_set
-        )
-
-        # Extract stoichiometries and compute mean values
-        (
-            min_sto_neg,
-            max_sto_neg,
-            min_sto_pos,
-            max_sto_pos,
-        ) = self._electrode_soh.get_min_max_stoichiometries(parameter_set)
-        mean_sto_neg = (min_sto_neg + max_sto_neg) / 2
-        mean_sto_pos = (min_sto_pos + max_sto_pos) / 2
-
-        # Calculate average voltage
-        positive_electrode_ocp = parameter_set["Positive electrode OCP [V]"]
-        negative_electrode_ocp = parameter_set["Negative electrode OCP [V]"]
-        try:
-            average_voltage = positive_electrode_ocp(
-                mean_sto_pos
-            ) - negative_electrode_ocp(mean_sto_neg)
-        except Exception as e:
-            raise ValueError(f"Error in average voltage calculation: {e}") from e
-
-        # Calculate the capacity estimate
-        approximate_capacity = theoretical_energy / average_voltage
-
-        return ParameterSet.evaluate_symbol(approximate_capacity, parameter_set)
+        # Calculate the theoretical capacity in the limit of low current
+        if self.pybamm_model.options["working electrode"] == "positive":
+            (
+                max_sto_p,
+                min_sto_p,
+            ) = self._electrode_soh_half_cell.get_min_max_stoichiometries(parameter_set)
+        else:
+            (
+                min_sto_n,
+                max_sto_n,
+                min_sto_p,
+                max_sto_p,
+            ) = self._electrode_soh.get_min_max_stoichiometries(parameter_set)
+
+        Q_p = LithiumIonParameters().p.prim.Q_init
+        theoretical_capacity = Q_p * (max_sto_p - min_sto_p)
```

Comment:
For the full cell calculation, this uses the positive electrode capacity exclusively; however, the negative electrode could be the limitation. I think we want to compare both electrode capacities and select the lower value for the theoretical energy capacity, right?

Reply:
I think the balancing is taken care of by pybamm's `get_min_max_stoichiometries`. These min and max values correspond to the values at 0 and 100% SOC, so it should be the same for each electrode, right?

Reply:
Ah, right. That simplifies things nicely - it might be worth adding a note in the docstring for future developers / users.